src/main/java/frc/robot/RobotContainer.java

```diff
@@ -103,15 +100,31 @@ public RobotContainer() {
     triggerSubsystem = new TriggerSubsystem();
     if (Constants.enabledSubsystems.limeLightEnabled
         && Constants.enabledSubsystems.drivetrainEnabled) {
-      limelightSubsystem = new LimelightSubsystem(drivetrain.getVisionMeasurementConsumer());
+      limelightSubsystem =
+          new LimelightSubsystem(drivetrain.getVisionMeasurementConsumer(), robotStateManager);
     } else {
       limelightSubsystem = null;
     }
+    if (Constants.enabledSubsystems.photonEnabled
+        && Constants.enabledSubsystems.drivetrainEnabled) {
+      photonSubsystem = new PhotonSubsystem(drivetrain.getVisionMeasurementConsumer());
+    } else {
+      photonSubsystem = null;
+    }
     if (Constants.enabledSubsystems.elvEnabled) {
       trapElvSubsystem = new TrapElvSubsystem();
     } else {
       trapElvSubsystem = null;
     }
+    if (Constants.enabledSubsystems.turretEnabled) {
+      if (Constants.enabledSubsystems.limeLightEnabled) {
+        turretSubsystem = new TurretSubsystem(robotStateManager, limelightSubsystem);
+      } else if (Constants.enabledSubsystems.photonEnabled) {
+        turretSubsystem = new TurretSubsystem(robotStateManager, photonSubsystem);
+      } else {
```

Comment:
We have two vision systems on the robot.  One for positioning, and the other for targeting.  Do we need a second photonSubsystem/limelightSubsystem here?

Reply:
I have compatibility for two photonvision cameras, but if we're planning to run both limelight and photonvision then I'll need to change more things around.